src/frontend/src/binder/expr/column.rs

```diff
@@ -75,7 +75,32 @@ impl Binder {
 
         // Try to find a correlated column in `upper_contexts`, starting from the innermost context.
         let mut err = ErrorCode::ItemNotFound(format!("Invalid column: {}", column_name));
-        for (i, (context, _)) in self.upper_subquery_contexts.iter().rev().enumerate() {
+
+        for (i, lateral_context) in self.lateral_contexts.iter().rev().enumerate() {
+            if lateral_context.is_visible {
+                let context = &lateral_context.context;
+                // input ref from lateral context `depth` starts from 1.
+                let depth = i + 1;
```

Comment:
Even for lateral subqueries, it seems using a depth used by normal subqueries is enough because the correlated input_ref  are in different depth levels.